.circleci/config.yml

```diff
@@ -0,0 +1,158 @@
+            pip install --upgrade pip twine wheel
+            pip install .[test] --upgrade
+
+      # Uncomment and adapt the next line to use a particular feature branch of OpenFisca-Core to run Circle CI tests
+      # - run:
+      #     name: Install a particular feature branch of OpenFisca-Core
+      #     command: |
+      #       . /tmp/venv/country-template/bin/activate
+      #       pip install --editable git+https://github.com/openfisca/openfisca-core.git@BRANCH_NAME#egg=OpenFisca-Core
+
+      - save_cache:
+          # PY_VERSION is set in the context
+          key: v1-{{ .Environment.PY_VERSION }}-dependencies-{{ .Environment.CIRCLE_SHA1 }}
+          paths:
+            - /tmp/venv/country-template
+
+  dependencies_python3:
+    <<: *dependencies
+    docker:
+      - image: python:3.6
+    environment:
+      CREATE_VENV: "python -m venv"
+
+  tests: &tests
+    working_directory: ~/country-template
+    docker:
+      - image: python:2.7.14
+
+    steps:
+      - restore_cache:
+          keys:
+            - v1-checkout-{{ .Environment.CIRCLE_SHA1 }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ .Environment.PY_VERSION }}-dependencies-{{ .Environment.CIRCLE_SHA1 }}
+
+      - run:
+          name: Run tests
+          command: |
+            . /tmp/venv/country-template/bin/activate
+            make test
+            . check-version-bump.sh
+
+  tests_python3:
+    <<: *tests
+    docker:
+      - image: python:3.6
+
+  deploy: &deploy
+    working_directory: ~/country-template
+    docker:
+      - image: python:2.7.14
+    environment:
+      PYPI_USERNAME: openfisca-bot
+      # PYPI_PASSWORD: this value is set in CircleCI's web interface; do not set it here, it is a secret!
+
+    steps:
+      - restore_cache:
+          keys:
+            - v1-checkout-{{ .Environment.CIRCLE_SHA1 }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ .Environment.PY_VERSION }}-dependencies-{{ .Environment.CIRCLE_SHA1 }}
+
+      - run:
+          name: Deploy (if version bump)
+          command: |
+            . /tmp/venv/country-template/bin/activate
+            . deploy-if-version-bump.sh
+
+  deploy_python3:
+    <<: *deploy
+    docker:
+      - image: python:3.6
+
+workflows:
+  version: 2
+  country-template:
+    jobs:
+      - checkout
+      - dependencies:
+          context: Py2
+          requires:
+            - checkout
+      - tests:
+          context: Py2
+          requires:
+            - dependencies
+      - dependencies_python3:
+          context: Py3
+          requires:
+            - checkout
+      - tests_python3:
+          context: Py3
+          requires:
+            - dependencies_python3
+      - deploy:
+          context: Py2
+          requires:
+            - tests
+            - tests_python3
+          filters:
+            branches:
+              only: master
+      - deploy_python3:
+          context: Py3
+          requires:
+            - tests
```

Comment:
This reads like “if I'm in `Py3` context, run `tests` (which is Python2)” 🤔 Is this normal?

Reply:
Got rid of contexts